packages/@storefront/sayt/src/previous-search-terms/index.ts

```diff
@@ -0,0 +1,37 @@
+import { provide, tag, Events, Selectors, Store, Tag } from '@storefront/core';
+import Sayt from '../sayt';
+
+@provide('previousSearch', (props) => props)
+@tag('gb-sayt-previous-search', require('./index.html'))
+class PreviousSearch {
+  props: PreviousSearch.Props = {
+    onClick: (query) => () => this.actions.search(query),
+  } as any;
+
+  constructor() {
+    const previousSearches = []
+    this.state = {...this.state, previousSearches}
+  }
+
+  init() {
+    this.flux.on(Events.ORIGINAL_QUERY_UPDATED, this.updatePreviousSearches);
+  }
+
+  updatePreviousSearches = (originalQuery: string) => {
```

Comment:
Lets change this from a function expression to a method declaration. ie.

```
updatePreviousSearches(originalQuery: string) {
  ...
}
```

Note that you'll also need to bind the method when registering it as a callback within `init()`. ie.

```
init() {
    this.flux.on(Events.ORIGINAL_QUERY_UPDATED, this.updatePreviousSearches.bind(this));
}
```

Reply:
Instead of binding in the `on` call, which is not easily testable, do this instead:
```js
constructor() {
  this.updatePreviousSearches = this.updatePreviousSearches.bind(this);
}
```

You'll see us setting function expressions in other components, but we're planning to move to the `bind` approach in the future.